platform/linux-generic/odp_ipsec.c

```diff
@@ -376,6 +377,40 @@ typedef struct {
 	uint8_t	iv[IPSEC_MAX_IV_LEN];
 } ipsec_state_t;
 
+/*
+ * Computes 64-bit seq number according to RFC4303 A2
+ */
+static inline uint64_t ipsec_compute_esn(ipsec_sa_t *ipsec_sa, uint32_t seq)
+{
+	uint32_t wintop_h, wintop_l, winbot_l, ws;
+	uint64_t seq64 = 0, wintop = 0;
+
+	wintop = odp_atomic_load_u64(&ipsec_sa->hot.in.wintop_seq);
+	wintop_l = wintop & 0xffffffff;
+	wintop_h = wintop >> 32;
+
+	ws = ipsec_sa->in.ar.win_size;
+	winbot_l = wintop_l - ws + 1;
+
+	/* case A: window is within one sequence number subspace */
+	if (wintop_l >= (ws - 1)) {
+		if (seq < winbot_l)
+			wintop_h++;
+	/* case B: window spans two sequence number subspaces */
+	} else {
+		if (seq >= winbot_l)
+			wintop_h--;
+	}
+
+	seq64 = (((uint64_t)wintop_h << 32) | seq);
```

Comment:
The outer parentheses are not needed here.

Reply:
Ack, will remove it.